InstallDotNetCore fails unexpectedly

I'm trying to use the new MultiFramework feature and seeing failures. This is currently blocking us and any help would be appreciated. 

```
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018: The "InstallDotNetCore" task failed unexpectedly. [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018: System.InvalidOperationException: Sequence contains no elements [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:  [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018: Server stack trace:  [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at System.Linq.Enumerable.First[TSource](IEnumerable`1 source) [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at Microsoft.DotNet.Arcade.Sdk.InstallDotNetCore.Execute() in /_/src/Microsoft.DotNet.Arcade.Sdk/src/InstallDotNetCore.cs:line 81 [C:\Users\username\.nuget\packages\microsoft.dotnet.a rcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs) [C:\Users\username\.nuget\packages\microsoft.d otnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg) [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.p roj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:  [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018: Exception rethrown at [0]:  [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg) [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\t ools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type) [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\T ools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at Microsoft.Build.Framework.ITask.Execute() [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\to ols\Tools.proj]
C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\InstallDotNetCore.targets(15,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [C:\Users\username\.nuget\packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.19264.13\tools\Tools.proj]
```

You can repro this by building [dotnet/wpf@dev/vatsanm/multi-frmwrk](https://github.com/dotnet/wpf/tree/dev/vatsanm/multi-frmwrk) like this; `build -pack -test /p:Platform=x64`


The framework install works fine (and the subsequent test runs function ok - on `x64` only) when installing manually by calling `eng\common\dotnet-install.cmd -runtime "dotnet" -version "3.0.0-preview6-27714-14" -architecture x64` (or any other specific version). 

/cc @chcosta 
/cc @rladuca, @miguep, @stevenbrix 

build logs: [Build.zip](https://github.com/dotnet/arcade/files/3181289/Build.zip)

--- a/src/Microsoft.DotNet.Arcade.Sdk/src/InstallDotNetCore.cs
+++ b/src/Microsoft.DotNet.Arcade.Sdk/src/InstallDotNetCore.cs
@@ -2,6 +2,7 @@ using Microsoft.Build.Evaluation;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 using NuGet.Versioning;
+using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
@@ -70,7 +71,7 @@ namespace Microsoft.DotNet.Arcade.Sdk
                                 else
                                 {
                                     var proj = Project.FromFile(VersionsPropsPath, new Build.Definition.ProjectOptions());
-                                    properties = proj.AllEvaluatedProperties.ToLookup(p => p.Name);
+                                    properties = proj.AllEvaluatedProperties.ToLookup(p => p.Name, StringComparer.OrdinalIgnoreCase );
                                 }
                             }
                             

